pydantic/_internal/_discriminated_union.py

```diff
@@ -12,6 +12,28 @@
 def apply_discriminator(
     schema: core_schema.CoreSchema, discriminator: str, definitions: dict[str, core_schema.CoreSchema] | None = None
 ) -> core_schema.CoreSchema:
+    """Applies the discriminator and returns a new core schema.
+
+    Args:
+        schema: The input schema.
+        discriminator: The name of the field which will serve as the discriminator.
+        definitions: A mapping of schema ref to schema.
+
+    Returns:
+        The new core schema.
+
+    Raises:
+        TypeError:
+            * If `discriminator` is used with invalid union variant.
+            * If `discriminator` is used with `Union` type with one variant.
+            * If `discriminator` value mapped to multiple choices.
```

Comment:
Do this look fine in the API documentation?

Reply:
not sure :). I can check. Do you have any suggestions?

Reply:
Ok, I've changed it and now it looks like:

![pic](https://github.com/pydantic/pydantic/assets/3122442/181790e2-78ff-40d5-8dac-54078d994648)